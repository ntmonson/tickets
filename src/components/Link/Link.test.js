/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

describe('Link', () => {
  test('renders with children correctly', () => {
    const component = renderer
      .create(
        <Link to="https://www.google.com/" onClick={() => {}}>
          <div>Test child div</div>
        </Link>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Handles the click correctly, no default prevented', () => {
    const component = renderer
      .create(
        <Link to="https://www.google.com/" onClick={() => {}}>
          <div>Test child div</div>
        </Link>,
      )
      .toJSON();

    component.props.onClick({ target: 'asdf' });

    expect(component).toMatchSnapshot();
  });
});
