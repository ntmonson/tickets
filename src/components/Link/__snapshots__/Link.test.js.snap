// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Handles the click correctly, no default prevented 1`] = `
<a
  href="https://www.google.com/"
  onClick={[Function]}
>
  <div>
    Test child div
  </div>
</a>
`;

exports[`Link renders with children correctly 1`] = `
<a
  href="https://www.google.com/"
  onClick={[Function]}
>
  <div>
    Test child div
  </div>
</a>
`;
